pacman/model/graphs/abstract_costed_partition.py

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) 2019-2020 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from six import add_metaclass
+from spinn_utilities.abstract_base import AbstractBase
+
+
+@add_metaclass(AbstractBase)
+class AbstractCostedPartition(object):
+
```

Comment:
Slots?

Reply:
total_sdram_requirements function???

Reply:
doing that causes a dimond inhertance. i think. will need to look more closly after merge

Reply:
so the issue here is that i dont know what other costed stuff we'll ever have. thats why ive left it as a separate thing